emily/handler/src/database/accessors.rs

```diff
@@ -696,6 +696,66 @@ pub async fn set_api_state(context: &EmilyContext, api_state: &ApiStateEntry) ->
 
 // Limits ----------------------------------------------------------------------
 
+/// Returns height of oldest stacks block anchored to given bitcoin block
+async fn get_oldest_stacks_block_for_bitcoin_block(
+    context: &EmilyContext,
+    bitcoin_height: u64,
+) -> Result<u64, Error> {
+    query_with_partition_key::<ChainstateByBitcoinHeightTableSecondaryIndex>(
+        context,
+        &bitcoin_height,
+        None,
+        None,
+    )
+    .await?
+    .0
+    .iter()
+    .min_by_key(|entry| entry.key.height)
+    .map(|entry| entry.key.height)
+    .ok_or(Error::NotFound)
+}
+
+async fn calculate_sbtc_left_for_withdrawals(
+    context: &EmilyContext,
+    rolling_withdrawal_blocks: Option<u64>,
+    rolling_withdrawal_cap: Option<u64>,
+) -> Result<u64, Error> {
+    let chaintip = get_api_state(context).await?.chaintip();
+    let bitcoin_tip = chaintip.bitcoin_height.ok_or(Error::NotFound)?;
+    let bitcoin_end_block = match rolling_withdrawal_blocks {
+        // We want to get oldest bitcoin block _included_ into rolling window, thus we substracting not
+        // window, but window - 1.
+        Some(window) => bitcoin_tip.saturating_sub(window - 1),
+        None => {
+            return Err(Error::NotFound);
+        }
+    };
+    let minimum_stacks_height_in_window =
+        get_oldest_stacks_block_for_bitcoin_block(context, bitcoin_end_block).await?;
+
+    let all_statuses_except_failed: Vec<_> = ALL_STATUSES
+        .iter()
+        .filter(|status| **status != Status::Failed)
+        .collect();
+
+    let mut withdrawals = vec![];
+    for status in all_statuses_except_failed {
+        let mut withdrawals_for_status =
+            get_all_withdrawal_entries_modified_from_height_with_status(
+                context,
+                status,
+                minimum_stacks_height_in_window,
+                None,
+            )
+            .await?;
+        withdrawals.append(&mut withdrawals_for_status);
```

Comment:
Since we only care about the sum, why not just keep track of the sum of the amounts?